### ISSUE TITLE
18 - Length of Tuple

### ISSUE THREAD

**Issue by JoaoVictor6** [type-challenges/type-challenges#13435]
This is my solution and I have some questions:
- is it necessary to a _readonly_?
- What is most legible, a literal type `unknown[]` or `Array<unknown>`

I hope someone sees and responds :)

```ts
type Length<T> = T extends Array<unknown> ? T['length'] : never
```

**Comment by samkahchiin** [comment]
> is it necessary to a readonly?

Yes. If you don't use `readonly`, you will get the following error:
```
Type 'readonly ["tesla", "model 3", "model X", "model Y"]' does not satisfy the constraint 'any[]'.
  The type 'readonly ["tesla", "model 3", "model X", "model Y"]' is 'readonly' and cannot be assigned to the mutable type 'any[]'
```

`tesla` and `spaceX` is readonly because of the `as const` (If you use VS Code, hover over the line and u'll see it appear as `const tesla: readonly  ["tesla", "model 3", "model X", "model Y"]`

**Comment by dany-fedorov** [comment]
> What is most legible, a literal type unknown[] or Array<unknown>

I understand that this is a matter of preference. 
My idea is that `Array<T>` form is more readable when there is a union passed as a parameter like `Array<string | number | boolean>`, because reading from left to right you can immediately see that this is an array, unlike with `(string | number | boolean)[]`. 
And the `T[]` syntax is better when there is just one type because it saves typing and reads ok.
